Add auto device status reporting to nRF Cloud (#10062)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

    * MCUboot child image files to properly access external flash on newer nRF9160DK versions.
    * An :file:`overlay_mcuboot_ext_flash.conf` file to enable MCUboot use of external flash.

* :ref:`nrf_cloud_rest_fota` sample:

  * Updated:

>     * Device status information, including FOTA enablement, is now sent nRF Cloud when the device connects.

Reviewer: ```suggestion
    * Device status information, including FOTA enablement, is now sent to nRF Cloud when the device connects.
```

    * Fixed an issue causing the A-GPS data download to be delayed until the RRC connection release.

* :ref:`modem_info_readme` library:

>   * Add :c:func:`modem_info_get_hw_version` function to obtain the HW version string using the AT%HWVERSION command.

Reviewer: ```suggestion
  * Added :c:func:`modem_info_get_hw_version` function to obtain the hardware version string using the ``AT%HWVERSION`` command.
```

  * Removed location events and event handlers.

* :ref:`lib_nrf_cloud_rest` library:

  * Added:

>     * Add the :c:func:`nrf_cloud_rest_device_status_message_send` function to send the device status information as an nRF Cloud device message.

Reviewer: ```suggestion
    * :c:func:`nrf_cloud_rest_device_status_message_send` function to send the device status information as an nRF Cloud device message.
```
